### PR TITLE
Don't show warning or critical errors for hosts in downtime

### PR DESCRIPTION
**Pull request by vizvayu**
Issue #339
This is working for me so far. May be a bit ugly so feel free to reject it if you prefer to code it better :)

### CODE REVIEW

**Comment by mohierf** [comment]
The idea is that if an host is in DOWNTIME, it is not counted anymore in DOWN or UNREACHABLE. Same for a service but for CRITICAL and WARNING. Right ?

But, if we retain this logic for synthesis popover, we should not include hosts and services that are acknowledged to remain consistent.

What do you think about it @maethor

File: module/helper.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def get_synthesis(self, elts):
         else:
             h['bi'] = 0
         for state in 'up', 'down', 'unreachable', 'pending':
-            h[state] = [i for i in hosts if i.state == state.upper()]
+            h[state] = [i for i in hosts if i.state == state.upper() and not ((state == 'unreachable' or state == 'down') and i.in_scheduled_downtime)]
```
Comment (maethor) [inline module/helper.py]: Why don't you simply add `and not i.in_scheduled_downtime`?

File: module/helper.py [hunk 2]
```diff
@@ -362,7 +362,7 @@ def get_synthesis(self, elts):
         else:
             s['bi'] = 0
         for state in 'ok', 'critical', 'warning', 'pending', 'unknown':
-            s[state] = [i for i in services if i.state == state.upper()]
+            s[state] = [i for i in services if i.state == state.upper() and not ((state == 'critical' or state == 'warning' or state == 'unknown') and i.in_scheduled_downtime)]
```
Comment (maethor) [inline module/helper.py]: Same comment.

**Comment by maethor** [comment]
The goal of the patch seems totally legit to me, but I don't see why we are still counting the OK. Can you clarify this point?

**Comment by mohierf** [comment]
I did not yet merged this PR because it changed the computing logic behind information displayed on header bar and on dashboard view ... but it also seems legit to me!

I think that I will merge this PR soon ... I am currently using this computing logic on my production environment ... :smile:

**Comment by mohierf** [comment]
PR #450 manages this.

**Pull request closed without merging**